cmake/legate_helper_functions.cmake

```
@@ -0,0 +1,395 @@
+    so_ext = {
+        "": "",
+        "Java": ".jar",
+        "Linux": ".so",
+        "Darwin": ".dylib",
+        "Windows": ".dll"
+    }[platform.system()]
+
+    def find_lib(libdir):
+        target = f"libhello{so_ext}*"
+        search_path = Path(libdir)
+        matches = [m for m in search_path.rglob(target)]
+        if matches:
+          return matches[0].parent
+        return None
+
+    return (
+        find_lib("@libdir@") or
+        find_lib(join(dirname(dirname(dirname(cn_path))), "lib")) or
+        find_lib(join(dirname(dirname(sys.executable)), "lib")) or
+        ""
+    )
+
+libpath: str = get_libpath()
+header: str = """@header_content@"""
+]=])
+
+  if (NOT SKBUILD OR NOT ${target}_DIR)
+    # if doing an editable C++ build prior to scikit-build install
+    # or doing a scikit-build from scratch, we need to generate install info
+    set(libdir ${CMAKE_BINARY_DIR}/legate_${target})
+    string(CONFIGURE "${install_info_in}" install_info @ONLY)
+    # this needs to go into the source tree for scikit-build install to work
+    file(WRITE ${CMAKE_SOURCE_DIR}/${target}/install_info.py "${install_info}")
+  endif()
+endfunction()
+
+function(legate_default_python_install target)
+  set(options)
+  set(one_value_args EXPORT)
+  set(multi_value_args)
+  cmake_parse_arguments(
+    LEGATE_OPT
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
+  )
+
+  if (NOT LEGATE_OPT_EXPORT)
+    message(FATAL_ERROR "Need EXPORT name for legate_default_python_install")
+  endif()
+
+  if (SKBUILD)
+    add_library(${target}_python INTERFACE)
+    add_library(legate::${target}_python ALIAS ${target}_python)
+    target_link_libraries(${target}_python INTERFACE legate::core legate::${target})
+
+    install(TARGETS ${target}_python
+            DESTINATION ${lib_dir}
+            EXPORT ${LEGATE_OPT_EXPORT})
+
+    legate_include_rapids()
+    rapids_export(
+      INSTALL ${target}_python
+      EXPORT_SET ${LEGATE_OPT_EXPORT}
+      GLOBAL_TARGETS ${target}_python
+      NAMESPACE legate::
+    )
+  endif()
+endfunction()
+
+function(legate_add_cpp_subdirectory dir)
+  set(options)
+  set(one_value_args EXPORT TARGET)
+  set(multi_value_args)
+  cmake_parse_arguments(
+    LEGATE_OPT
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
+  )
+
+  if (NOT LEGATE_OPT_EXPORT)
+    message(FATAL_ERROR "Need EXPORT name for legate_default_install")
+  endif()
+
+  if (NOT LEGATE_OPT_TARGET)
+    message(FATAL_ERROR "Need TARGET name for Legate package")
+  endif()
+  # abbreviate for the function
+  set(target ${LEGATE_OPT_TARGET})
+
+  legate_include_rapids()
+
+  rapids_find_package(legate_core CONFIG
+          GLOBAL_TARGETS legate::core
+          BUILD_EXPORT_SET ${LEGATE_OPT_EXPORT}
+          INSTALL_EXPORT_SET ${LEGATE_OPT_EXPORT})
+
+  if (SKBUILD)
+    if (NOT DEFINED ${target}_ROOT)
+      set(${target}_ROOT ${CMAKE_SOURCE_DIR}/build)
```

```suggestion
      set(${target}_ROOT ${CMAKE_BINARY_DIR}/legate_${target})
```
This needs to be the same as the add_subdirectory call

---

This is a a shortcut so people's `pip install` will "just work" if they follow a standard directory naming scheme. When this runs, `CMAKE_BINARY_DIR` will be `_skbuild` rather than the C++ build directory.

---

It's admittedly hacky...